sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala

```diff
@@ -261,5 +261,18 @@ class DataFrameNaFunctionsSuite extends QueryTest with SharedSQLContext {
     assert(out1(3).get(2).asInstanceOf[Double].isNaN)
     assert(out1(4) === Row("Amy", null, null))
     assert(out1(5) === Row(null, null, null))
+
+    // Replace with null
+    val out2 = input.na.replace("name", Map(
+      "Bob" -> "Bravo",
+      "Alice" -> null
```

Comment:
I saw you allow a replacement like (k: Double, null). Can you also add a test for such replacement? Thanks.